sources/api/schnauzer/src/v2/template.rs

```diff
@@ -67,15 +74,52 @@ impl TemplateFrontmatter {
     pub fn extension_requirements(&self) -> impl Iterator<Item = ExtensionRequirement> + '_ {
         self.required_extensions
             .as_ref()
-            .map(|requirements| Box::new(requirements.iter()) as Box<dyn Iterator<Item = _>>)
-            .unwrap_or_else(|| Box::new(std::iter::empty()) as Box<dyn Iterator<Item = _>>)
+            .map(|requirements| Box::new(requirements.iter()) as Box<dyn Iterator<Item = _> + Send>)
+            .unwrap_or_else(|| Box::new(std::iter::empty()) as Box<dyn Iterator<Item = _> + Send>)
             .map(|(extension_name, extension_requirements)| {
                 ExtensionRequirement::from_template_requirements(
                     extension_name,
                     extension_requirements,
                 )
             })
     }
+
+    /// Performs parse-time validations of a `TemplateFrontmatter`.
+    fn validate(&self) -> Result<()> {
+        self.ensure_no_helper_names_collide()?;
+        Ok(())
+    }
+
+    /// Ensures imported helper names do not collide.
+    ///
+    /// While settings are namespaced in the templates (`settings.extension.key_of_interest`), helpers cannot be
+    /// similarly namespaced due to a bug in the `handlebars` crate which fails to parse helpers containing a `.`
```

Comment:
Is there an upstream issue we can link to here?

Reply:
So far as I know, I discovered this issue. Planning to open an issue but I also kinda want to see if I can submit a PR for it.

Reply:
https://github.com/sunng87/handlebars-rust/issues/595